### PR TITLE
Feature/skale 2694 mainet launch

### PR DESCRIPTION
**Pull request by DimaStebaev**
Changes to support mainnet launch:
- don't pay bounty before network launch
- schains creation before token minting

### CODE REVIEW

File: contracts/SkaleDKG.sol [hunk 1]
```diff
@@ -77,24 +51,19 @@ contract SkaleDKG is Permissions {
...
     }
 
     struct KeyShare {
         bytes32[2] publicKey;
         bytes32 share;
     }
 
-    uint constant private _P = 21888242871839275222246405745257275088696311157297823662689037894645226208583;
-
     uint constant private _G2A = 10857046999023057135944570762232829481370756359578518086990519993285655852781;
```
Comment (payvint) [inline contracts/SkaleDKG.sol]: Why did not you move it to library?

File: contracts/SkaleDKG.sol [hunk 2]
```diff
@@ -364,25 +319,35 @@ contract SkaleDKG is Permissions {
         SchainsInternal schainsInternal = SchainsInternal(
             _contractManager.getContract("SchainsInternal")
         );
-        channels[groupIndex].active = true;
 
-        delete channels[groupIndex].completed;
-        for (uint i = 0; i < channels[groupIndex].broadcasted.length; i++) {
-            delete _data[groupIndex][i];
+        channels[groupIndex].active = true;
+        while(channels[groupIndex].completed.length > 0) {
```
Comment (payvint) [inline contracts/SkaleDKG.sol]: I think we could change it to `delete channels[groupIndex].completed`

File: contracts/SkaleDKG.sol [hunk 3]
```diff
@@ -364,25 +319,35 @@ contract SkaleDKG is Permissions {
...
+            channels[groupIndex].broadcasted.pop();
         }
-        delete channels[groupIndex].broadcasted;
-        channels[groupIndex].broadcasted = new bool[](schainsInternal.getNumberOfNodesInGroup(groupIndex));
-        channels[groupIndex].completed = new bool[](schainsInternal.getNumberOfNodesInGroup(groupIndex));
-        delete channels[groupIndex].publicKey.x.a;
-        delete channels[groupIndex].publicKey.x.b;
-        channels[groupIndex].publicKey.y.a = 1;
-        delete channels[groupIndex].publicKey.y.b;
+        for (uint i = 0; i < schainsInternal.getNumberOfNodesInGroup(groupIndex); ++i) {
```
Comment (payvint) [inline contracts/SkaleDKG.sol]: What's the reason to change it that way?

**Review by payvint — CHANGES_REQUESTED** [review]

File: contracts/SkaleDKG.sol (hunk 3 above)
Comment (DimaStebaev) [reply contracts/SkaleDKG.sol]: It reduces code size a little bit because assigned value is not store in the code.

**Review by payvint — APPROVED** [review]
AWESOME!

**Pull request merged**